### PR TITLE
support multiple roles

### PR DESCRIPTION
**Pull request by mulby**
Use different roles for different schemas loaded into Vertica. This creates three categories of roles: standard, restricted, and business_intelligence_team. It also adds the concept of "restricted columns" which are only visible to the users with the `restricted` role.

FYI @HassanJaveed84

### CODE REVIEW

**Comment by mulby** [comment]
OK - my manual tests passed:

* Ran a warehouse load into production_mulby
* Ran an event load
* Ran an otto database import

This is just waiting on review to merge.

File: edx/analytics/tasks/common/vertica_load.py [hunk 1]
```diff
@@ -428,6 +428,30 @@ def copy_data_table_from_target(self, cursor):
...
             )
 
+    @property
+    def restricted_columns(self):
+        return []
+
+    def create_access_policies(self, connection):
+        cursor = connection.cursor()
+        for column in self.restricted_columns:
+            restricted_roles_param = luigi.Parameter(is_list=True, config_path={'section': 'vertica-export', 'name': 'restricted_roles'})
```
Comment (brianhw) [inline edx/analytics/tasks/common/vertica_load.py]: This looks like it is required to be found in the config file.  Any reason for that?  I guess that's okay if one is declaring a table as having a restricted column.  Alternative is to have restricted roles be empty, so only dbadmin will be enabled for seeing the column.

File: edx/analytics/tasks/common/vertica_load.py [hunk 2]
```diff
@@ -428,6 +428,30 @@ def copy_data_table_from_target(self, cursor):
...
+CREATE ACCESS POLICY ON {schema}.{table} FOR COLUMN {column}
+CASE WHEN {expression} THEN {column}
+ELSE 'Restricted'
+END
+ENABLE;""".format(schema=self.schema, table=self.table, column=column, expression=expression)
+            try:
+                cursor.execute(statement)
+            except Exception:
+                # This is expected to fail if the access policy already exists.
+                # TODO: check if this is an "expected" failure, or an unexpected one.
```
Comment (brianhw) [inline edx/analytics/tasks/common/vertica_load.py]: Shouldn't this be figured out before merging to master?

File: edx/analytics/tasks/warehouse/load_internal_reporting_events.py [hunk 3]
```diff
@@ -1219,6 +1219,11 @@ class PruneEventPartitionsInVertica(EventRecordLoadDownstreamMixin, SchemaManage
     # create queries to delete the excess partitions.
     earliest_date = None
 
+    roles = luigi.Parameter(
```
Comment (brianhw) [inline edx/analytics/tasks/warehouse/load_internal_reporting_events.py]: nit: I had to look at code elsewhere to realize this was overwriting the standard definition of roles, and that vertica_roles derives from roles.  Might help to have a brief comment saying that we are overwriting the
parameter in order to restrict access.

**Review by brianhw — COMMENTED** [review]
LGTM overall.  One question about error handling, however (and nits on doc).

File: edx/analytics/tasks/common/vertica_load.py (hunk 1 above)
Comment (mulby) [reply edx/analytics/tasks/common/vertica_load.py]: Good point, I've made the parameter optional!

File: edx/analytics/tasks/common/vertica_load.py (hunk 2 above)
Comment (mulby) [reply edx/analytics/tasks/common/vertica_load.py]: Yeah, I'll run a test now.

File: edx/analytics/tasks/warehouse/load_internal_reporting_events.py (hunk 3 above)
Comment (mulby) [reply edx/analytics/tasks/warehouse/load_internal_reporting_events.py]: Good idea. Done.

File: edx/analytics/tasks/common/vertica_load.py [hunk 4]
```diff
@@ -428,6 +429,37 @@ def copy_data_table_from_target(self, cursor):
...
+            log.debug(statement)
+            try:
+                cursor.execute(statement)
+            except QueryError as query_error:
+                # This is expected to fail if the access policy already exists.
+                expected_error = 'Access policy for COLUMN "{column}" already exists on "{table}"'.format(
+                    column=column,
+                    table=self.table
+                )
+                if query_error.error_response.mssage == expected_error:
```
Comment (brianhw) [inline edx/analytics/tasks/common/vertica_load.py]: Is this a typo? (mssage?)

**Review by brianhw — APPROVED** [review]
Fix the typo, otherwise LGTM.  👍

**Review by tobz — APPROVED** [review]
LGTM. 👍

**Pull request merged**